tesseract-server/src/handlers/logic_layer/geoservice.rs

```diff
@@ -31,23 +32,16 @@ pub fn query_geoservice(
     geoservice_query: &GeoserviceQuery,
     geo_id: &str
 ) -> Result<Vec<GeoServiceResponseJson>, Error> {
-    let mut query_url: String;
+    let base_url = Url::parse(geoservice_url.clone()).unwrap();
 
-    if &geoservice_url[geoservice_url.len()-1..] == "/" {
-        match geoservice_query {
-            GeoserviceQuery::Neighbors => query_url = format!("{}neighbors/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Children => query_url = format!("{}relations/children/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Parents => query_url = format!("{}relations/parents/{}", geoservice_url, geo_id),
-            _ => return Err(format_err!("This type of geoservice query is not yet supported"))
-        }
-    } else {
-        match geoservice_query {
-            GeoserviceQuery::Neighbors => query_url = format!("{}/neighbors/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Children => query_url = format!("{}/relations/children/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Parents => query_url = format!("{}/relations/parents/{}", geoservice_url, geo_id),
-            _ => return Err(format_err!("This type of geoservice query is not yet supported"))
-        }
-    }
+    let join_str = match geoservice_query {
+        GeoserviceQuery::Neighbors => format!("{}neighbors/{}", geoservice_url, geo_id),
+        GeoserviceQuery::Children => format!("{}relations/children/{}", geoservice_url, geo_id),
+        GeoserviceQuery::Parents => format!("{}relations/parents/{}", geoservice_url, geo_id),
+        _ => return Err(format_err!("This type of geoservice query is not yet supported"))
+    };
+
+    let query_url = base_url.join(&join_str).unwrap().as_str().to_string();
```

Comment:
does it work if you remove `to_string()`, but pass to client as `&query_url`?

Reply:
![Screen Shot 2019-08-14 at 11 53 27 AM](https://user-images.githubusercontent.com/6521281/63036019-47bf6280-be8a-11e9-9cd7-5403f0fe94e3.png)

This is what I get when I remove the `to_string()`